lib/axlsx/workbook/worksheet/auto_filter/auto_filter.rb

```diff
@@ -48,15 +49,51 @@ def add_column(col_id, filter_type, options = {})
       columns.last
     end
 
-    # actually performs the filtering of rows who's cells do not
-    # match the filter.
+    # Performs the sorting of the rows based on the sort_state conditions. Then it actually performs
+    # the filtering of rows who's cells do not match the filter.
     def apply
       first_cell, last_cell = range.split(':')
       start_point = Axlsx.name_to_indices(first_cell)
       end_point = Axlsx.name_to_indices(last_cell)
       # The +1 is so we skip the header row with the filter drop downs
       rows = worksheet.rows[(start_point.last + 1)..end_point.last] || []
 
+      # the sorting of the rows if sort_conditions are available.
+      if !sort_state.sort_conditions.to_a.empty? && self.sort_on_generate
```

Comment:
```suggestion
      if !sort_state.sort_conditions.to_a.empty? && sort_on_generate
```

`self.` will trigger an offense after #285